### PR TITLE
JwtX5tValidator: minor improvements

### PR DESCRIPTION
**Pull request by nenaraab**
- avoid nullpointer when calling `JwtX5tValidator.validate(null)`
- Logger.ERROR is not required, log is written with `ValidationResults.createInvalid(...)`
- removed audience check as documented in latest version of proofOfPossession Doc -> see #2-check-certificate-thumbprint-x5t

### CODE REVIEW

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java [hunk 1]
```diff
@@ -29,11 +30,10 @@
 public class JwtX5tValidator implements Validator<Token> {
 
 	private static final Logger LOGGER = LoggerFactory.getLogger(JwtX5tValidator.class);
-	private final OAuth2ServiceConfiguration config;
-
+	static final String VALIDATION_FAILED = "Certificate cnf token validation failed";
```
Comment (liga-oz) [inline java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:33]: maybe better "Certificate thumbprint validation failed"

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java [hunk 2]
```diff
@@ -29,11 +30,10 @@
...
 
 	public JwtX5tValidator(OAuth2ServiceConfiguration config) {
-		this.config = config;
+		Assertions.assertNotNull(config, "config must not be null");
```
Comment (liga-oz) [inline java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:36]: I would prefer a bit more detailed message like "OAuth2ServiceConfiguration must not be null"

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java [hunk 3]
```diff
@@ -51,26 +51,23 @@ public JwtX5tValidator(OAuth2ServiceConfiguration config) {
 	 */
 	@Override
 	public ValidationResult validate(Token token) {
+		if (token == null) {
+			return ValidationResults.createInvalid("No token passed to extract cnf thumbprint");
```
Comment (liga-oz) [inline java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:55]: No token passed to X5tValidator?

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java [hunk 4]
```diff
@@ -51,26 +51,23 @@ public JwtX5tValidator(OAuth2ServiceConfiguration config) {
...
-				if (token.getAudiences().size() == 1 && token.getAudiences().contains(config.getClientId())) {
-					return ValidationResults.createValid();
-				}
-				LOGGER.error("Audience validation failed -> \"aud\": {} != \"clientid\": \"{}\"", token.getAudiences(), config.getClientId());
-			} else {
-				LOGGER.error("Thumbprint validation failed -> x5t from token: \"{}\" != thumbprint from client certificate: \"{}\"", tokenX5t,
-						clientCertificateX5t);
-			}
+		if (tokenX5t == null) {
+			return ValidationResults.createInvalid("Token doesn't provide cnf thumbprint");
```
Comment (liga-oz) [inline java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:60]: change message to "Token doesn't contain certificate thumbprint confirmation method" to align with [RFC8705](https://datatracker.ietf.org/doc/html/rfc8705#section-3.1) terminology

File: java-security/src/test/java/com/sap/cloud/security/servlet/IasTokenAuthenticatorX509Test.java [hunk 5]
```diff
@@ -80,14 +80,14 @@ void validateRequest_validTokenValidX5t_noCertificate() {
 		TokenAuthenticationResult response = cut.validateRequest(httpRequest, HTTP_RESPONSE);
 
 		assertThat(response.getUnauthenticatedReason())
-				.contains("Error during token validation: Certificate validation failed");
+				.contains("Error during token validation: Client certificate missing");
```
Comment (liga-oz) [inline java-security/src/test/java/com/sap/cloud/security/servlet/IasTokenAuthenticatorX509Test.java:83]: adapt messages accordingly to the ones in `JwtX5tValidator`

File: java-security/src/test/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidatorTest.java [hunk 6]
```diff
@@ -50,94 +49,64 @@ static void beforeAll() throws IOException {
 				.withClientId("myClientId")
 				.build();
 		CUT = new JwtX5tValidator(configuration);
```
Comment (liga-oz) [inline java-security/src/test/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidatorTest.java:51]: a test with JwtX5tValidator initialization with null would be good to have 
```java
new JwtX5tValidator(null)
```

**Review by liga-oz — CHANGES_REQUESTED** [review]
Just some suggestions to improve logging messages.

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java (hunk 3 above)
Comment (nenaraab) [reply java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:55]: jwtx5tvalidator is internal info and already part of the logger name.

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java (hunk 2 above)
Comment (nenaraab) [reply java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:36]: updated to "No token passed to validate certificate thumbprint"

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java (hunk 3 above)
Comment (nenaraab) [reply java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:55]: updated to "No token passed to validate certificate thumbprint"

File: java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java (hunk 2 above)
Comment (nenaraab) [reply java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtX5tValidator.java:36]: rephrased it to "Service configuration must not be null" as on other places.

**Review by liga-oz — APPROVED** [review]
💯

**Pull request merged**